src/ripple/core/impl/Config.cpp

```diff
@@ -479,6 +479,18 @@ Config::loadFromString(std::string const& fileContents)
 
     std::string strTemp;
 
+    if (getSingleSection(secConfig, SECTION_NETWORK_ID, strTemp, j_))
+    {
+        if (strTemp == "main")
+            NETWORK_ID = 0;
+        else if (strTemp == "testnet")
+            NETWORK_ID = 1;
+        else if (strTemp == "devnet")
+            NETWORK_ID = 2;
+        else
+            NETWORK_ID = beast::lexicalCastThrow<uint32_t>(strTemp);
+    }
```

Comment:
I noticed this section is similar to [the `OverlayImpl.cpp` version](https://github.com/XRPLF/rippled/blob/e32bc674aa2a035ea0f05fe43d2f301b203f1827/src/ripple/overlay/impl/OverlayImpl.cpp#L1615-L1628); is that something that can be updated to be consistent or shared in some way so this switch doesn't have to be in multiple places?

Reply:
I also saw this but didn't want to touch OverlayImpl. Feel free to suggest an additional commit with appropriate tests to remove this redundant code.